src/components/ArticlesList.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from "react";
+import { getArticles } from "../api";
+import ArticleCard from "./ArticleCard";
+import "./ArticlesList.css";
+
+const ArticlesList = () => {
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
```

Comment:
remember to setIsLoading to true, incase re-rending happens after error occurring